src/features/defi/providers/osmosis/components/OsmosisManager/Deposit/components/Deposit.tsx

```diff
@@ -0,0 +1,375 @@
+   * NOTE: This calculation may need to be modified to support pools with unequal weights at a later date. The Osmosis frontend does not currently do this.
+   */
+  const calculateAllocations = useCallback(
+    (
+      inputAsset: Asset,
+      inputAssetAmount: string,
+    ): { allocationFraction: string; shareOutAmount: string } => {
+      const ret = { allocationFraction: '0', shareOutAmount: '0' }
+      if (!state || !state.poolData) {
+        return ret
+      }
+      const poolAssets = state?.poolData?.pool_assets
+
+      let { assetReference: inputAssetReference } = fromAssetId(inputAsset.assetId)
+      if (inputAssetReference === ASSET_REFERENCE.Osmosis) inputAssetReference = 'uosmo'
+
+      if (!poolAssets) return ret
+      const poolAssetIndex = poolAssets.findIndex(asset =>
+        asset.token.denom.toLowerCase().includes(inputAssetReference.toLowerCase()),
+      )
+      if (poolAssetIndex < 0) return ret
+
+      const inputAssetAmountBaseUnits = bnOrZero(inputAssetAmount)
+        .pow(10, bnOrZero(inputAsset.precision))
+        .toString()
+
+      const poolTotalSharesBaseUnits = state?.poolData?.total_shares?.amount
+      if (!poolTotalSharesBaseUnits) {
+        moduleLogger.error(`Could not get total shares for pool ${state?.poolData?.id}`)
+        return ret
+      }
+
+      const poolAssetAmountBaseUnits = poolAssets[poolAssetIndex]?.token.amount
+
+      if (bnOrZero(poolAssetAmountBaseUnits).eq(bn(0))) return ret
+
+      /** allocation_fraction = input_asset_amount / (pool_asset_amount + pool_asset_amount)
+       * This represents the fraction of the pool that the user will own after once the liquiditt
+       * has been provided
+       */
+      const allocationFraction = bnOrZero(inputAssetAmountBaseUnits)
+        .div(bnOrZero(poolAssetAmountBaseUnits).plus(inputAssetAmountBaseUnits))
+        .toString()
+
+      /** share_out_amount = total_pool_shares * (input_asset_amount / pool_asset_amount)
+       * This represents the number of pool shares that the user will receive in exchange
+       * for the provided liquidity
+       */
+      const shareOutAmountBaseUnits = bnOrZero(poolTotalSharesBaseUnits)
+        .multipliedBy(
+          bnOrZero(inputAssetAmountBaseUnits).dividedBy(bnOrZero(poolAssetAmountBaseUnits)),
+        )
+        .toFixed(0)
+        .toString()
+
+      ret.allocationFraction = allocationFraction
+      ret.shareOutAmount = shareOutAmountBaseUnits
+      return ret
+    },
+    [state],
+  )
+
+  const handleContinue = useCallback(
+    async (formValues: DepositValues) => {
+      if (
+        !(
+          state &&
+          contextDispatch &&
+          userAddress &&
+          opportunity &&
+          underlyingAsset0 &&
+          underlyingAsset1
+        )
+      )
+        return
+
+      const asset0AmountBaseUnits = bnOrZero(formValues.cryptoAmount1)
```

Comment:
Two hard things in software engineering, mainly for grep OCD purposes
```suggestion
      const asset0AmountBaseUnit = bnOrZero(formValues.cryptoAmount1)
```